File: example/iOS/App/DemoLibrary.m
//
//  DemoLibrary.m
//  Example
//
//  Created by shangkun on 2022/7/26.
//  Copyright © 2022 pengfei.zhou. All rights reserved.
//

#import "DemoLibrary.h"
#import "DemoPlugin.h"

@implementation DemoLibrary

- (void)load:(DoricRegistry *)registry {
    [registry registerNativePlugin:DemoPlugin.class withName:@"demoPlugin"];
}

@end

File: example/android/app/src/main/java/pub/doric/android/DemoLibrary.java
package pub.doric.android;

import pub.doric.DoricComponent;
import pub.doric.DoricLibrary;
import pub.doric.DoricRegistry;

@DoricComponent
public class DemoLibrary extends DoricLibrary {

    @Override
    public void load(DoricRegistry registry) {
        registry.registerNativePlugin(DemoPlugin.class);
    }
}